File: ruby/zero-to-n.rb
def main
  print "Number n: ".strip
  n = gets.strip.to_f
  current_number = 0
  negative = true
  while current_number <= n
    if negative
      puts -current_number
      negative = false
    else
      puts current_number
      negative = true
    end
    current_number += 1
  end
end

main

File: python/zero-to-n.py
def main():
    n = int(float(input("Number n: ")))
    if n >= 0:
        current_number = 0
        negative = True
        while current_number <= n:
            if negative:
                negative = False
                print(-current_number)
            else:
                negative = True
                print(current_number)
            current_number += 1


if __name__ == '__main__':
    main()
